lstchain/reco/dl1_to_dl2.py

```diff
@@ -787,14 +785,21 @@ def calc_source_dependent_parameters(data, expected_src_pos_x_m, expected_src_po
     src_dep_params['expected_src_x'] = expected_src_pos_x_m
     src_dep_params['expected_src_y'] = expected_src_pos_y_m
 
-    src_dep_params['dist'] = np.sqrt((data['x'] - expected_src_pos_x_m) ** 2 + (data['y'] - expected_src_pos_y_m) ** 2)
-
+    if 'lhfit_x' in data.keys():
+        x = 'lhfit_x'
+        y = 'lhfit_y'
+        psi = 'lhfit_psi'
+    else:
+        x = 'x'
+        y = 'y'
+        psi = 'psi'
+    src_dep_params['dist'] = np.sqrt((data[x] - expected_src_pos_x_m) ** 2 + (data[y] - expected_src_pos_y_m) ** 2)
     disp, miss = camera_to_shower_coordinates(
         expected_src_pos_x_m,
         expected_src_pos_y_m,
-        data['x'],
-        data['y'],
-        data['psi'])
+        data[x],
+        data[y],
+        data[psi])
 
     src_dep_params['time_gradient_from_source'] = data['time_gradient'] * np.sign(disp) * -1
     src_dep_params['skewness_from_source'] = data['skewness'] * np.sign(disp) * -1
```

Comment:
`time_gradient` and `skewness` (and the sign of those parameters) are computed based on the standard hillas parameter (`x`, `y`, `psi`), then `lhfit_time_gradient` and `lhfit_length_assymmetry` are computed based on lhfit parameters (`lhfit_x`, `lhfit_y`, `lhfit_psi`), right? If so, this sign correction should be done using the base parameters.

Reply:
Right, I will change this

Reply:
Now `dist` and `alpha` are computed using LHfit parameters, but the sign of `time_gradient` is not corrected. Is this on purpose?

Reply:
The sign of the `skewness` and `time gradient` are corrected line 795-796 of the current version. But this is now in non-edited code as it uses a different disp from Hillas x and y, while `dist` and `alpha` use the one obtained with the lhfit quantities.